tools/clang/lib/AST/HlslTypes.cpp

```diff
@@ -68,6 +68,29 @@ bool IsHLSLVecMatType(clang::QualType type) {
   return false;
 }
 
+clang::RecordDecl *getRecordTypeFromParameterType(clang::QualType ParamTy) {
```

Comment:
I think the name of this function should be more like "GetRecordTypeFromNodeObjectType" instead (and ParamTy could be ObjectTy).  The type being passed in is some node object type with a record template parameter.  This function does not operate on a "ParameterType".